### PR TITLE
Util tweaks

### PR DESCRIPTION
**Pull request by gabriel**
- Move rand util stuff into rand.go
- RandString changed to RandomID and uses 256 bits
- Move string util into strings.go

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/updater-hackers

File: util/rand_test.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+func TestRandString(t *testing.T) {
+	s, err := RandomID("prefix.")
+	t.Logf("Rand string: %s", s)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !strings.HasPrefix(s, "prefix.") {
+		t.Errorf("Invalid prefix: %s", s)
+	}
+	if len(s)-len("prefix.") != 52 {
```
Comment (chrisnojima) [inline util/rand_test.go]: does this actually work if the random number encodes with padding (which then gets removed)?

Comment (gabriel) [inline util/rand_test.go]: Good catch, yeah the padding remove should only happen from the returned base32 string. Fixed with test.

File: util/rand_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+func TestRandString(t *testing.T) {
+	s, err := RandomID("prefix=")
+	t.Logf("Rand string: %s", s)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !strings.HasPrefix(s, "prefix=") {
+		t.Errorf("Invalid prefix: %s", s)
+	}
+	if len(s)-len("prefix.") != 52 {
```
Comment (chrisnojima) [inline util/rand_test.go]: even with the above fix won't this fail sometimes? sometimes the base32 encoding will have padding / won't (or is that not true). It seems like if it does sometimes and doesn't sometimes, then the length of the random string (ignoring prefix) will be variable length (cause we strip it sometimes) so it won't always be 52 long. like sometimes it'll be 51 and still be working fine

Comment (gabriel) [inline util/rand_test.go]: ah.. the random id is always 256 bits (as suggested by https://gist.github.com/tqbf/be58d2d39690c3b366ad)... this means the padding will always be the same

**Pull request merged**